classes/question_ui_renderer.php

```diff
@@ -355,27 +369,62 @@ private function resolve_placeholders(): void {
             $cleanoption = $parts[1] ?? 'clean';
 
             if (!isset($this->placeholders[$key])) {
+                // No value for this placeholder, so we just remove the PI.
                 $pi->parentNode->removeChild($pi);
+                continue;
+            }
+
+            $rawvalue = $this->placeholders[$key];
+            if (strcasecmp($cleanoption, 'clean') == 0) {
+                // Allow HTML, but clean using Moodle's clean_text to prevent XSS.
+                $element = $this->xml->createDocumentFragment();
+                if (!$this->append_html_fragment($element, clean_text($rawvalue))) {
+                    throw new coding_exception('clean_text produced invalid HTML');
```

Comment:
Mir gefallen Exceptions nicht, wenn dadurch das ganze Quiz kaputt gemacht werden kann. Auch sollte eine invalide User-Eingabe nicht den Attempt in Mitleidenschaft ziehen. Ich finde es daher besser, wenn wir gar nichts oder einen Fehlertext einfügen, evtl. ein debugging call.